prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -238,11 +236,29 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 	 * construct pkt_entry. The actual receiving buffer
 	 * posted to device starts from pkt_entry->wiredata.
 	 */
-	pkt_entry->pkt_size = rxe->iov[0].iov_len - sizeof(struct efa_rdm_pke);
-
+	pkt_entry->pkt_size = rxe->total_len - sizeof *pkt_entry;
 	pkt_entry->ope = rxe;
 	rxe->state = EFA_RDM_RXE_MATCHED;
 
+	if (ep->user_info->mode & FI_MSG_PREFIX) {
+		err = ofi_iov_locate(rxe->iov, rxe->iov_count, ep->msg_prefix_size, &rx_iov_index, &rx_iov_offset);
+		if (OFI_UNLIKELY(err)) {
+			EFA_WARN(FI_LOG_CQ, "ofi_iov_locate failure: %s (%d)\n", fi_strerror(-err), -err);
+			return err;
+		}
+
+		assert(rx_iov_index < rxe->iov_count);
+		assert(rx_iov_offset < rxe->iov[rx_iov_index].iov_len);
+
+		pkt_entry->payload = (char *) rxe->iov[rx_iov_index].iov_base + rx_iov_offset;
+		pkt_entry->payload_mr = rxe->desc[rx_iov_index];
+		ofi_consume_iov(rxe->iov, &rxe->iov_count, ep->msg_prefix_size);
```

Comment:
I realize we can use ofi_consume_iov_desc https://github.com/ofiwg/libfabric/blob/main/src/iov.c#L63  so we can shift iov and desc together. Then we don't have to use ofi_iov_locate to get the rx_iov_index and rx_iov_offset

Reply:
Accordingly, for this line we also need to use ofi_consume_iov_desc https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/efa_rdm_ope.c#L60-L62